js/adapt-contrib-scoring.js

```diff
@@ -83,9 +83,10 @@ class Scoring extends Backbone.Controller {
    * @fires Adapt#scoring:update
    */
   update() {
-    const updateSubsets = !this._queuedChanges?.length
+    const queuedChanges = [...new Set(this._queuedChanges)];
```

Comment:
Is it better to check if the [queue](https://github.com/adaptlearning/adapt-contrib-scoring/blob/3a4b504fc7fe75039a2ed909c72edfa866f7ff1a/js/adapt-contrib-scoring.js#L382) already contains the model before adding, rather than removing duplicates later?

Reply:
It makes no difference with arrays at this size.

It's difficult to quantify which is more performant.

It may be cheaper to remove duplicates as the list grows, as at each addition there would be fewer duplicate comparisons and the list would remain as short as possible. However, the overhead of performing a duplicate comparison each time might cost more than performing a single duplicate check on a large set of mostly non-duplicates.

It entirely depends on the quantity of duplicate and non-duplicate items, the frequency of additions and whether it's more important to add quickly now and process more slowly later, or add slowly now and process more quickly later and then also if ram usage is important.

etc